### PR TITLE
TPCClusterFinder: Enable calibrating parameters.

### PR DESCRIPTION
**Pull request by fweig**
This PR replaces the constants in the tpc cluster finder by flags that can be set at runtime.

### CODE REVIEW

File: GPU/GPUTracking/TPCClusterFinder/TPCCFCalibration.h [hunk 1]
```diff
@@ -42,11 +44,20 @@ struct TPCPadGainCorrectionStepNum<unsigned short> {
 
 struct TPCCFCalibration {
  public:
+  // Calibration parameters for cluster finder
+  float qmaxCutoff = 3.f;
+  float qtotCutoff = 0.f;
+  float extendInnerChargeThreshold = 0.f;
+  float minSplitChargeNum = 1.f;
+  float noiseSuppressionEpsilon = 10.f;
```
Comment (davidrohr) [inline GPU/GPUTracking/TPCClusterFinder/TPCCFCalibration.h]: I don't quite understand why you have copies of these values here. Cannot you use directly the values from the GPUParam.rec?

File: GPU/GPUTracking/Base/GPUSettingsList.h [hunk 2]
```diff
@@ -43,6 +43,11 @@ AddOption(MaxTrackQPt, float, 1.f / GPUCA_MIN_TRACK_PT_DEFAULT, "", 0, "required
 AddOption(SearchWindowDZDR, float, 2.5, "", 0, "Use DZDR window for seeding instead of vertex window")
 AddOption(TrackReferenceX, float, 1000.f, "", 0, "Transport all tracks to this X after tracking (disabled if > 500, auto = 1000)")
 AddOption(tpcZSthreshold, float, 2.0f, "", 0, "Zero-Suppression threshold")
+AddOption(tpcCFqmaxCutoff, float, 3.0f, "", 0, "Cluster Finder rejects cluster with qmax below this threshold")
+AddOption(tpcCFqtotCutoff, float, 0.f, "", 0, "Cluster Finder rejects cluster with qtot below this threshold")
+AddOption(tpcCFinnerThreshold, float, 0.f, "", 0, "Cluster Finder extends cluster if inner charge above this threshold")
+AddOption(tpcCFminSplitNum, float, 1.f, "", 0, "Minimum number of split charges in a cluster for the cluster to be marked as split")
+AddOption(tpcCFnoiseSuppressionEpsilon, float, 10.f, "", 0, "Cluster Finder: Difference between peak and charge for the charge to count as a minima during noise suppression")
```
Comment (davidrohr) [inline GPU/GPUTracking/Base/GPUSettingsList.h]: We can save some constant space here if we use unsigned char instead of float.

**Review by davidrohr — COMMENTED** [review]
Just have 2 minor comments.
And did you check that the performance on the GPU is identical to before?

File: GPU/GPUTracking/TPCClusterFinder/TPCCFCalibration.h (hunk 1 above)
Comment (fweig) [reply GPU/GPUTracking/TPCClusterFinder/TPCCFCalibration.h]: Totally missed that the GPUSettingsRec struct is also available on GPU. Which makes a lot of sense of course. Parameters are now accessed from there directly instead.

File: GPU/GPUTracking/Base/GPUSettingsList.h (hunk 2 above)
Comment (fweig) [reply GPU/GPUTracking/Base/GPUSettingsList.h]: Fixed.

**Comment by fweig** [comment]
Just pushed an update to address your points.

Performance looks virtually identical to `dev`.

**Pull request merged**